extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerRecorder.java

```diff
@@ -107,15 +107,21 @@ public void configureRegistries(MicrometerConfig config,
             }
         }
 
+        List<AutoCloseable> autoCloseables = new ArrayList<>();
+
         // Base JVM Metrics
         if (config.checkBinderEnabledWithDefault(() -> config.binder.jvm)) {
             new ClassLoaderMetrics().bindTo(Metrics.globalRegistry);
-            new JvmHeapPressureMetrics().bindTo(Metrics.globalRegistry);
+            JvmHeapPressureMetrics jvmHeapPressureMetrics = new JvmHeapPressureMetrics();
+            jvmHeapPressureMetrics.bindTo(Metrics.globalRegistry);
+            autoCloseables.add(jvmHeapPressureMetrics);
             new JvmMemoryMetrics().bindTo(Metrics.globalRegistry);
             new JvmThreadMetrics().bindTo(Metrics.globalRegistry);
             new JVMInfoBinder().bindTo(Metrics.globalRegistry);
             if (ImageMode.current() == ImageMode.JVM) {
-                new JvmGcMetrics().bindTo(Metrics.globalRegistry);
+                JvmGcMetrics jvmGcMetrics = new JvmGcMetrics();
+                jvmGcMetrics.bindTo(Metrics.globalRegistry);
+                autoCloseables.add(jvmGcMetrics);
```

Comment:
I wonder if I would add all the listeners to a List and then close them if they are Closeable or AutoCloseable. That might be more future proof.

Reply:
I understand where the desire for defensiveness against such OOMs is coming from, but in this case, I think it's overkill.

In reality, this issue **was**  being flagged by IntelliJ (as the auto-closable was never being closed), we just didn't pay attention - and we absolutely should next time :)